lib/internal/crypto/mac.js

```diff
@@ -68,17 +78,46 @@ async function hmacGenerateKey(algorithm, extractable, keyUsages) {
     extractable);
 }
 
-function hmacImportKey(
+async function kmacGenerateKey(algorithm, extractable, keyUsages) {
+  const { name } = algorithm;
+  let { length } = algorithm;
+
+  if (length === undefined)
+    length = NumberParseInt(StringPrototypeSubstring(name, 4), 10);
```

Comment:
This definitely feels very magic-y and won't look correct to readers unless they know that `name` is an enum with values that happen to have the same format ... how about a more explicit mapping like this?

```suggestion
  length ??= {
    __proto__: null,
    KMAC128: 128,
    KMAC256: 256
  }[name];
```

Reply:
Have to agree. Either that or a comment that explains it.